spec/ruby/core/hash/new_spec.rb

```diff
@@ -46,4 +46,22 @@
       Hash.new({ unknown: true }).default.should == { unknown: true }
     end
   end
+
+  ruby_version_is "3.4" do
+    it "accepts a capacity: argument" do
+      Hash.new(5, capacity: 42).default.should == 5
+      Hash.new(capacity: 42).default.should == nil
+      (Hash.new(capacity: 42) { 1 }).default_proc.should_not == nil
+    end
+
+    it "ignores negative capacity" do
+      -> { Hash.new(capacity: -42) }.should_not raise_error
```

Comment:
I suppose this is debatable, but I chose to behave like `String.new(capacity:)` for consistency.